File: PlayerStage/psTest/src/psTest.cpp
#include <stdio.h>
#include <stdlib.h>
#include <math.h>
#include <libplayerc++/playerc++.h>
#include <iostream>

int
main(int argc, const char **argv)
{
  int i;  

  try
  {
    using namespace PlayerCc;

    // Sortu bezeroa eta konektatu zerbitzariarekin.
    PlayerClient bezeroa("localhost", 6665);
    // Sortu laser interfazea eta harpidetu bezeroarekin
    LaserProxy laserra(&bezeroa, 0);
    // Sortu position2d interfazea eta harpidetu bezeroarekin
    Position2dProxy robota(&bezeroa, 0);
    
    std::cout << bezeroa << std::endl;
    
    robota.SetMotorEnable (true);

    for (i = 0; i < 10; i++)
      bezeroa.Read();

    
    int count = laserra.GetCount();
   
    while (1)
      { 
	bezeroa.Read();
	for (i = 0; i < count; i++)
	  {
	    std::cout << laserra.GetRange(i) << ", ";
	  }
	std::cout << "-----------------------------------" << std::endl;
	robota.SetSpeed(0, 0.2);
      }
  }
  catch (PlayerCc::PlayerError & e)
    {
      std::cerr << e << std::endl;
      return -1;
    }
    
}

File: PlayerStage/psTest/scripts/psTest.py
import numpy as np
import math
from playercpp import *
from time import sleep


def main():

    bezeroa = PlayerClient("localhost", 6665)
    laserra = LaserProxy(bezeroa, 0)
    robota = Position2dProxy(bezeroa, 0)

    print(bezeroa)

    robota.SetMotorEnable(True)

    for i in range(0,10):
        bezeroa.Read()
    count = laserra.GetCount()

    while 1:
        bezeroa.Read()
        for i in range(0, count):
            print(i, "Reading: ", laserra.GetRange(i))
        print("-----------------------------------------")
        robota.SetSpeed(0, 0.2)
        sleep(0.1)
if __name__ == "__main__":
    main()
